newrelic/core/application.py

```diff
@@ -1671,6 +1700,18 @@ def internal_agent_shutdown(self, restart=False):
 
         self.stop_data_samplers()
 
+        # Finishes collecting environment plugin information
+        # if this has not been completed during harvest
+        # lifetime of the application
+
+        while self.configuration.package_reporting.enabled and self._env_send:
+            try:
+                module_info = next(self.plugins)
+                self.modules.append(module_info)
+            except StopIteration:
+                self._env_send = False
+                self._active_session.send_loaded_modules(self.modules)
```

Comment:
It's possible there are no modules to send in this case as well. Just in case, clear the modules after sending.
```suggestion
                if self.modules:
                   
self._active_session.send_loaded_modules(self.modules)
self.modules = []
```